spark/v3.0/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -272,6 +273,17 @@ public void invalidateTable(Identifier ident) {
       load(ident).first().refresh();
     } catch (org.apache.iceberg.exceptions.NoSuchTableException ignored) {
       // ignore if the table doesn't exist, it is not cached
+    } catch (TableUUIDMismatchException e) {
+      try {
+        if (cacheEnabled && !isPathIdentifier(ident)) {
+          ((CachingCatalog) icebergCatalog).invalidate(buildIdentifier(ident));
+          load(ident);
+        } else {
+          throw e;
+        }
+      } catch (org.apache.iceberg.exceptions.NoSuchTableException ignored) {
+        // ignore if the table doesn't exist, it is not cached
+      }
```

Comment:
Would it possibly be better to use some mechanism other than more exceptions for control flow?

Reply:
Fixed. Thanks @kbendick

Reply:
Sorry just noticed I used a thumbs down reaction when I meant thumbs up!